pdg/src/info.rs

```diff
@@ -1,31 +1,83 @@
 use crate::Graphs;
+use crate::graph::{Graph,NodeId,Node,NodeKind};
 use std::fmt::{self, Debug, Display, Formatter};
+use std::collections::HashMap;
 
 /// Information generated from the PDG proper that is queried by static analysis.
 ///
-/// Eventually this will include information about what kinds of [`Node`]s the [`Node`] is an ancestor of,
-/// as well as its ability to be used as a `&mut`.
+/// Includes information about what kinds of [`Node`]s the [`Node`] flows to,
+/// and eventually will also include its ability to be used as a `&mut`.
 ///
 /// [`Node`]: crate::graph::Node
-#[derive(Hash, Clone, PartialEq, Eq, Debug)]
-pub struct NodeInfo {}
+#[derive(Hash, Clone, PartialEq, Debug)]
+pub struct NodeInfo {
+    flows_to: Flows,
+}
+
+/// Contains information about what kinds of [`Node`]s the [`Node`] flows to.
+/// Load and store kinds contain both 
+///
+/// [`Node`]: crate::graph::Node.
+#[derive(Debug, Hash, Clone, Copy,PartialEq,Default)]
+pub struct Flows {
+    load: Option<NodeId>,
+    store: Option<NodeId>,
+    pos_offset: Option<NodeId>,
+    neg_offset: Option<NodeId>,
+}
 
 impl Display for NodeInfo {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
         write!(f, "")
     }
 }
 
+fn init_flows(n_id: NodeId, n: &Node) -> Flows {
```

Comment:
I think this would be better off as a method in `Flows` like `Flows::new`

Reply:
Also, does it need the whole `Node` or just the `NodeKind`?

Reply:
Changed on both counts.